### PR TITLE
TP2000-608 Unit tests for api list views

### PR DESCRIPTION
**Pull request by a-gleeson**
# TP2000-608  Unit tests for api list views

## Why
Previously the ReadOnlyModelViewSet views did not have any units tests, this story ensures that this coverage has been addressed

## What
Unit testing API view sets

### CODE REVIEW

File: workbaskets/tests/test_views.py [hunk 1]
```diff
@@ -51,7 +51,10 @@ def test_workbasket_create_user_not_logged_in_dev_sso_disabled(client, settings)
     settings.ENV = "dev"
     settings.SSO_ENABLED = False
     settings.LOGIN_URL = reverse("login")
-    settings.MIDDLEWARE.remove("authbroker_client.middleware.ProtectAllViewsMiddleware")
+    if "authbroker_client.middleware.ProtectAllViewsMiddleware" in settings.MIDDLEWARE:
```
Comment (gabelton) [inline workbaskets/tests/test_views.py:54]: Why is this change needed?

File: additional_codes/urls.py [hunk 2]
```diff
@@ -11,7 +11,11 @@
     views.AdditionalCodeViewSet,
     basename="additionalcode",
 )
-api_router.register(r"additional_code_types", views.AdditionalCodeTypeViewSet)
+api_router.register(
+    r"additional_code_types",
+    views.AdditionalCodeTypeViewSet,
+    basename="additionalcodetype",
```
Comment (gabelton) [inline additional_codes/urls.py:17]: Do we need to add basename args? It gets the same name from the view queryset, I think

File: workbaskets/tests/test_views.py (hunk 1 above)
Comment (a-gleeson) [reply workbaskets/tests/test_views.py:54]: When running tests locally this wasn't in the test middleware and would throw an exception. So I just added a check so it wouldn't fail.

File: additional_codes/urls.py (hunk 2 above)
Comment (a-gleeson) [reply additional_codes/urls.py]: Not needed, the ones I added are now removed

**Review by gabelton — APPROVED** [review]
